index.js

```diff
@@ -216,6 +215,7 @@ NodeListIterator.prototype.next = function () {
  * @return {Node} The heap's new minimum node.
  */
 function cut(node, parent, minNode, compare) {
+  node.parent = undefined;
```

Comment:
It also makes incorrect reference and therefor decreaseKey() logic became broken.